guides/migrate_to_keras_3.py

```diff
@@ -0,0 +1,963 @@
+    print(layer.trainable_variables)
+```
+
+`UserWarning: The model does not have any trainable weights.
+  warnings.warn("The model does not have any trainable weights.")`
+"""
+
+"""
+How you fix it:
+
+use `self.add_weight()` method or opt for a `keras.Variable` instead. If you
+are currently using `tf.variable`, you can switch to `keras.Variable`.
+
+"""
+
+
+class MyCustomLayer(keras.layers.Layer):
+    def __init__(self, units):
+        super().__init__()
+        self.units = units
+
+    def build(self, input_shape):
+        input_dim = input_shape[-1]
+        self.w = self.add_weight(
+            shape=[input_dim, self.units],
+            initializer="zeros",
+        )
+        self.b = self.add_weight(
+            shape=[
+                self.units,
+            ],
+            initializer="zeros",
+        )
+
+    def call(self, inputs):
+        return keras.ops.matmul(inputs, self.w) + self.b
+
+
+layer = MyCustomLayer(3)
+data = np.random.uniform(size=[3, 3])
+model = keras.models.Sequential([layer])
+model.compile(optimizer="adam", loss="mse")
+model.fit(data, data)
+# Verify that the variables are now being tracked
+for layer in model.layers:
+    print(layer.trainable_variables)
+
+"""
+### None entries are not allowed as part of nested (e.g. list/tuples) tensor
+arguments in `Layer.call()`, nor as part of `call()`'s nested return values.
+
+If the `None` in the argument is intentional and serves a specific purpose,
+ensure that the argument is optional and structure it as a separate parameter.
+For example, consider defining the `call` method with optional argument.
+
+The following snippet of code will reproduce the error.
+
+```python
+class CustomLayer(keras.layers.Layer):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, inputs):
+      return  inputs["foo"]
+
+layer = CustomLayer()
+inputs = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {
+        "baz": None,
+    },
+}
+layer(inputs)
+```
+"""
+
+"""
+How you fix it:
+
+Replace `None` with a value or Keras Tensor
+"""
+
+
+class CustomLayer(keras.layers.Layer):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, inputs):
+        output = inputs["foo"]
+        return None
+
+
+layer = CustomLayer()
+inputs = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {
+        "baz": keras.Input(shape=(1,), name="bar"),
+    },
+}
+layer(inputs)
+
+
+"""
+Or
+
+Define the call method with an optional argument. Here is an example of this
+fix:
+"""
+
+
+class CustomLayer(keras.layers.Layer):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, input_1, input_2=None):
+        output_ = input_1["foo"]
+        output_2 = input_2
+        return None
+
+
+layer = CustomLayer()
+input_1 = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {},
+}
+baz = None
+layer(input_1, input_2=baz)
+
+
+"""
+### Random seed behavior changes between Keras 2 and Keras 3.
```

Comment:
I'll touch up this section a bit after merging.

Reply:
SG!!